buildSrc/src/main/java/org/mozilla/fenix/gradle/tasks/ApkSizeTask.kt

```diff
@@ -0,0 +1,126 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.gradle.tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import org.json.JSONArray
+import org.json.JSONObject
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Paths
+
+/**
+ * Gradle task for determining the size of APKs and logging them in a perfherder compatible format.
+ */
+open class ApkSizeTask : DefaultTask() {
+    /**
+     * Name of the build variant getting built.
+     */
+    @Input
+    var variantName: String? = null
+
+    /**
+     * List of APKs that get build for the build variant.
+     */
+    @Input
+    var apks: List<String>? = null
+
+    @TaskAction
+    fun logApkSize() {
+        val apkSizes = determineApkSizes()
+        val json = buildPerfherderJson(apkSizes)
+
+        println("PERFHERDER_DATA: $json")
+    }
+
+    private fun determineApkSizes(): Map<String, Long> {
+        val basePath = listOf(
+            "${project.projectDir}", "build", "outputs", "apk", variantName
+        ).joinToString(File.separator)
+
+        return requireNotNull(apks).associateWith { apk ->
+            val path = Paths.get("$basePath${File.separator}$apk")
+            Files.size(path)
```

Comment:
Do we want to fail the build if we can't determine size for whatever reason (e.g. an `IOException`) or should we show the size as `unknown` then?

Reply:
Ah, good point. We probably never want to fail the build for this. I'll catch this.